File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/events/BotDeathEvent.java
package dev.robocode.tankroyale.server.events;

import lombok.Value;

/**
 * Event sent when a bot has been killed.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BotDeathEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bot id of the victim that got killed */
	int victimId;
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/event/BotDeathEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.BotId

/** Event sent when a bot has been killed. */
class BotDeathEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bot id of the victim that got killed */
    val victimId: BotId,
) : Event()
